### ISSUE TITLE
Shutdown Request and Exit Notification do not work

### ISSUE THREAD

**Issue by hediet** [Microsoft/vscode-languageserver-node#96]
Shutdown requests and exit notifications do not work as expected or not at all when using nodejs as server. 
When using .Net as server, shutdown requests are sent (and received by the server) only if there is no breakpoint in the visual studio code instance that debugs the client extension. Exit notifications are not sent. However, due to the experimental state of my .Net server I cannot guarantee that the bug is there.
It feels like this bug is an timing issue.

### Steps to reproduce:
* Setup example language server as described [here](https://code.visualstudio.com/docs/extensions/example-language-server).
* Add the following code before `connection.listen();` in `server.ts`:
```typescript
connection.onShutdown(() => {
	fs.appendFile('C:/...some-absolute-path.../message.txt', '.shutdown');
});
connection.onExit(() => {
	fs.appendFile('C:/...some-absolute-path.../message.txt', '.exit');
});
```
Add the following code to the `onInitialize` handler:
```typescript
fs.appendFile('C:/...some-absolute-path.../message.txt', '.init');
```
* Start visual studio code, wait some seconds
* Close visual studio code

### Expected behavior
The file `message.txt` contains `.init.shutdown.exit`

### Actual behavior
The file `message.txt` contains `.init`

**Comment by dbaeumer** [comment]
I noticed that lately has well. Something must have changed in how we are shutting down VS Code. Need to investigate where this is coming from. Need @alexandrudima for this.

**Comment by alexandrudima** [comment]
@dbaeumer New electron version? I haven't changed anything in there since months

**Comment by dbaeumer** [comment]
@alexandrudima and me sat down together and fix it. Was an issue in VS Code itself. The fix in in VS Code 1.7 or in VS Code 1.7-insider.
